File: Arrays/Multi-dimensional.js
const prompt=require('prompt-sync')()

// 1] print elements of arrays in a matrix form





// let arr=[[1,2,3],[4,5,6],[7,8,9]];
// for(let i=0;i<arr.length;i++){
//     for(let j=0;j<arr[i].length;j++){
//         process.stdout.write(`${arr[i][j]} `);
//     }
//     console.log();
    
// }












// 2] take input from user and print in matrix form;

// let rows =Number(prompt("Enter number of rows "));
// let colums=Number(prompt("enter number of colums "));


// let arr= new Array(rows);
// for(let i=0;i<arr.length;i++){
//      arr[i]= new Array(colums);
// }
// for(let i=0;i<arr.length;i++){
//     for(let j=0;j<arr[i].length;j++){
//         arr[i][j]=Number(prompt(`enter elemts`));
//     }
// }
// console.log(arr);

















// 3] sum of diagonal elements of a n*n matrix





// let martix =[
//     [1,2,3],
//     [4,5,6],
//     [7,8,9]
// ];
// let sum=0;
// for(let i=0;i<martix.length;i++){
//     for(let j=0;j<martix[i].length;j++){
//         if(i==j || i+j==martix.length-1){
//             sum+=martix[i][j]
//         }
        
//     }
// }
// console.log(sum);













// 5] rotate image :






// let  matrix = [[1,2,3],[4,5,6],[7,8,9]]



// for(let i=0;i<matrix.length;i++){
//     for(let j=i;j<matrix[i].length;j++){
//         let temp=matrix[i][j];
//         matrix[i][j]=matrix[j][i];
//         matrix[j][i]=temp;
//     }
// }

















// 6] Spiral matrix :





// let matrix =[
//     [1,2,3],
//     [4,5,6],
//     [7,8,9]
// ];
// let ans=[];
// let minrow=0;
// let maxrow=matrix.length-1;
// let mincol=0;
// let maxcol=matrix[0].length-1;
// let m=matrix.length,n=matrix[0].length

// while(ans.length<m*n){
//    for(let i=minrow;i<=maxcol;i++){
//     ans.push(matrix[minrow][i]);
//    }
//    minrow++;
//    for(let i=minrow;i<=maxrow;i++){
//     ans.push(matrix[i][maxcol]);
//    }
//    maxcol--
   
//    for(let i=maxcol ; i>=mincol;i--){
//     ans.push(matrix[maxrow][i])
//    }
//    maxrow--
//    for(let i=maxrow;i>=minrow;i--){
//     ans.push(matrix[i][mincol])
//    }
//    mincol++

// }
// console.log(ans)





let x=123;
let copy=x;
let rem,rev=0;
while(x>0){
    rem=x%10;
    rev=(rev*10)+rem;
    x=Math.floor(x/10)

    
}
console.log(typeof(rev));

File: Arrays/multi-dimensional.java
// 1. Diagonal sum of a matrix


// import java.util.Scanner;
// public class Main{
//     public static void main(String[] args){
//         Scanner sc=new Scanner(System.in);
//         System.out.print("Enter row and col ");
//         int row=sc.nextInt();
//         int col=sc.nextInt();
//         int[][] arr=new int[row][col];
//         for(int i=0;i<row;i++){
//             for(int j=0;j<col;j++){
//                 arr[i][j]=sc.nextInt();
//             }
//         }
//         int leftSum=0,rightSum=0;
//         for(int i=0;i<row;i++){
//             for(int j=0;j<col;j++){
//                 if(i==j) leftSum+=arr[i][j];
//                 if(i+j==arr.length-1) rightSum+=arr[i][j];
//             }
//         }
//         System.out.print("Right sum = "+rightSum);
//          System.out.print("Left sum = "+leftSum);
        
        
//     }
// }





// 2. Transpose of a matrix


// import java.util.Scanner;
// public class Main{
//     public static void main(String[] args){
//         Scanner sc=new Scanner(System.in);
//         System.out.print("Enter row and col ");
//         int row=sc.nextInt();
//         int col=sc.nextInt();
//         int[][] arr=new int[row][col];
//         for(int i=0;i<row;i++){
//             for(int j=0;j<col;j++){
//                 arr[i][j]=sc.nextInt();
//             }
//         }
//         for(int i=0;i<row;i++){
//             for(int j=0;j<arr[i].length;j++){
//                 System.out.print(arr[i][j]+" ");
                
//             }
//             System.out.println();
//         }
        
//         System.out.println("Transpose Matrix : ");
        
//         int[][] ans=new int[col][row];
//         for(int i=0;i<ans.length;i++){
//             for(int j=0;j<ans[i].length;j++){
//                 ans[i][j]=arr[j][i];
//             }
//         }
        
//          for(int i=0;i<ans.length;i++){
//             for(int j=0;j<ans[i].length;j++){
//                 System.out.print(ans[i][j]+" ");
                
//             }
//             System.out.println();
//         }
        
        
        
//     }
// }




// 3. Rotate a matrix by 90 degree

// import java.util.Scanner;
// import java.util.Arrays;
// public class Main{
//     public static void reverse(int start,int end,int[] arr){
//         while(start<end){
//             int temp=arr[start];
//             arr[start]=arr[end];
//             arr[end]=temp;
//             start++;
//             end--;
          
            
//         }
      
//     }
//     public static void main(String[] args){
//         Scanner sc=new Scanner(System.in);
//         int n=sc.nextInt();
//         int[][] matrix=new int[n][n];
//         for(int i=0;i<matrix.length;i++){
//             for(int j=0;j<matrix[i].length;j++){
//                 matrix[i][j]=sc.nextInt();
//             }
//         }
//         for(int i=0;i<matrix.length;i++){
//             for(int j=0;j<matrix[i].length;j++){
//                 System.out.print(matrix[i][j]+" ");
//             }
//             System.out.println();
//         }
//         System.out.println();
//         for(int i=0;i<matrix.length;i++){
//             for(int j=i;j<matrix[i].length;j++){
//                 int temp=matrix[i][j];
//                 matrix[i][j]=matrix[j][i];
//                 matrix[j][i]=temp;
//             }
//         }
//         for(int i=0;i<matrix.length;i++){
          
//                 reverse(0,matrix[i].length-1,matrix[i]);                
            
//         }
//         for(int i=0;i<matrix.length;i++){
//             for(int j=0;j<matrix[i].length;j++){
//                 System.out.print(matrix[i][j]+" ");
//             }
//             System.out.println();
//         }
//     }
// }